src/oss-metadata/MetadataTool.cs

```diff
@@ -55,7 +56,7 @@ public MetadataTool(IHttpClientFactory httpClientFactory) : base(httpClientFacto
             try
             {
                 // Use reflection to find the correct downloader class
-                BaseProjectManager? projectManager = ProjectManagerFactory.CreateProjectManager(purl, httpClientFactory);
+                BaseProjectManager? projectManager = ProjectManagerFactory.CreateProjectManagerWithProvider(purl, new NuGetProvider(), httpClientFactory);
```

Comment:
This change is very messy.  If CreateManagerWithProvider always takes a `new NuGetProvider()` with no parameters why not just construct it in the old create project manager method only when needed?

Reply:
This was a quick first try solution for me. I have already refactored a lot of this into a new branch. But the change is pretty large scoped so I wasn't sure if I should include it with this PR https://github.com/jpinz/OSSGadget/tree/jupinzer/projectmanager_provider

Reply:
This other version looks much better.  I think it makes more sense to do it holistically in one PR unless you need the nuget specific version merged imminently for some reason.  It looks like the new interface declared in this PR are changed right away with the other branch.

Reply:
I published the other PR as a draft if you want to check it out.

Reply:
The new PR is out.